### PR TITLE
Make video sizing consistent with images

### PR DESCRIPTION
**Pull request by robintown**
Type: defect

Closes https://github.com/vector-im/element-web/issues/20072.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Make video sizing consistent with images ([\#8102](https://github.com/matrix-org/matrix-react-sdk/pull/8102)). Fixes vector-im/element-web#20072.

Preview: https://pr8102--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by ara4n** [comment]
This looks almost perfect having tried it out on the netlify preview - thank you!!

From a quick look, the code looks good too.

I have one thought: which is that I think it was an accidental bug that 'large' thumbnails were ever allowed to expand wider than 800px.  Originally the timeline width was fixed to 800px and only got changed to support arbitrarily wide layouts quite late in the day... and whoever did it let the thumbnails then expand to be arbitrarily wide, which is unfortunate given the typical max thumbnail width from the server is 800px.  As a result, any window wider than 800px would cause the thumbnail to be upscaled, making it look awful (especially if it's something like a screenshot).  Therefore, I suggest we fix this better late than never and cap the 'large' thumbnail size to fit within 800x600 rather than allowing arbitrary width.

We also have a problem that on retina displays (as most are, these days), a 800px wide thumbnail from the server is going to be at half-resolution when scaled to fit a width of 800 device-independent pixels - and so look unnecessarily fuzzy.  Not sure what the right solution is there (asking the server to generate 1600x1200 'thumbnails' seems a bit crazy?). Alternatively, giving up on the thumbnail entirely and pulling down the full-res image (if it's some massive photo) could be overkill too.  Perhaps it's time to bite the bullet and configure synapse to gen 1600x1200 thumbs for nice crispy 'large' thumbnails (and also to check that appropriate retina thumbnails are made for the non-large thumbnail sizing too).

**Comment by turt2live** [comment]
we've also changed the calculations with the optional "smaller thumbnails" user setting - previously they were width constrained (and still are for "large"), but for small thumbnails we height-constrained them instead, leading to interesting results.

Retina requirements should be something we handle a bit better in today's `Media` handler, but it's possible the image/video code was so risky we didn't set it up properly.

**Comment by ara4n** [comment]
>  previously they were width constrained (and still are for "large")

i am pretty sure thumbnails are not width constrained when in "large" mode? hence my req to cap to 800px

**Review by turt2live — APPROVED** [review]
Code generally seems fine, though design might want to consider the thumbnail sizing constraints.

**Comment by turt2live** [comment]
> > previously they were width constrained (and still are for "large")
> 
> i am pretty sure thumbnails are not width constrained when in "large" mode? hence my req to cap to 800px

The shape is defined by the width, which is how tall videos/images become obscenely tall (because they fall under the maximum).

**Comment by ara4n** [comment]
okay, terminology confusion over the word 'constrained' then :)

TL;DR: this is fine to merge; i don't think we need design input on the thumbnail sizing as it hasn't changed... but please can the 'large' thumbnail size be capped to fit into 800x600 rather than just being capped to be 600px high?

**Comment by robintown** [comment]
Large widths are now capped. @ara4n is your evening review of this your way of saying you'd like this in the RC?

**Comment by turt2live** [comment]
I'd recommend we don't for internal reasons - will share internally.

**Comment by ara4n** [comment]
don't need this in the RC; i'm only working in the evening because i miss hacking on EW and can't sleep :P
